proxy/http/HttpTransact.cc

```diff
@@ -1423,7 +1423,12 @@ HttpTransact::PPDNSLookup(State *s)
     HTTP_INCREMENT_DYN_STAT(http_total_parent_marked_down_count);
     s->parent_params->markParentDown(&s->parent_result, s->txn_conf->parent_fail_threshold, s->txn_conf->parent_retry_time);
     // DNS lookup of parent failed, find next parent or o.s.
-    find_server_and_update_current_info(s);
+    if (find_server_and_update_current_info(s) == HttpTransact::HOST_NONE) {
```

Comment:
Looks good.
There is a slight inconsistency from the other caller `HandleCacheOpenReadMiss()`, which checks `s->parent_result.result == PARENT_FAIL` instead.

Reply:
I'm not sure here why we check the dst_addr.valid() in the original code.